src/software/new_geom/util/intersects_test.cpp

```diff
@@ -0,0 +1,414 @@
+#include "software/new_geom/util/intersects.h"
+
+#include <gtest/gtest.h>
+
+#include "software/new_geom/rectangle.h"
+#include "software/new_geom/triangle.h"
+
+TEST(IntersectsTest, polygon_segment_intersecting_on_edge)
```

Comment:
Can we add a test for a `Segment` completely contained within a `Polygon`?

Reply:
Added test `segment_contained_in_polygon`. (Realized this is inconsistent with the behaviour of 2D shape intersects under discussion, subject to change)